tests/unit/ml_handlers/test_google_gemini.py

```diff
@@ -0,0 +1,105 @@
+import os
+import pytest
+import pandas as pd
+from unittest.mock import patch
+
+from .base_ml_test import BaseMLAPITest
+
+GEMINI_API_KEY = os.environ.get('GOOGLE_GENAI_API_KEY')
+
+
+@pytest.mark.skipif(GEMINI_API_KEY is None, reason='Missing API key!')
+class TestGeminiHandler(BaseMLAPITest):
+    """Test Class for Google Gemini (Bard) API handler"""
+
+    def setup_method(self):
+        """Setup test environment, creating a project"""
+        super().setup_method()
+        self.run_sql("create database proj")
+
+    def test_invalid_model_parameter(self):
+        """Test for invalid Gemini model parameter"""
+        self.run_sql(
+            f"""
+            CREATE MODEL proj.test_google_invalid_model
+            PREDICT answer
+            USING
+                engine='google_gemini',
+                column='question',
+                model='non-existing-gemini-model',
+                api_key='{GEMINI_API_KEY}';
+            """
+        )
+        with pytest.raises(Exception):
+            self.wait_predictor("proj", "test_google_invalid_model")
+
+    @pytest.mark.skip(reason="This test is failing as no error is being thrown")
+    def test_unknown_model_argument(self):
```

Comment:
@someshfengde This test fails. the reason being there is no argument checker in the handler. I marked to skip until we fix

I suggest you use `pydantic`  to validate handler arguments on create - see [ here](https://github.com/mindsdb/mindsdb/blob/61f389ac7e5843459544b9860b5085229e800bd4/mindsdb/integrations/handlers/rag_handler/rag_handler.py#L61l) for an example

Reply:
I had added this into the create method previously i think this should be alright 

https://github.com/mindsdb/mindsdb/blob/d22b743e733d1475202ea4970e726fd68a1e66bb/mindsdb/integrations/handlers/google_gemini_handler/google_gemini_handler.py#L35C1-L35C1